crates/core/tedge_actors/src/servers/message_boxes.rs

```diff
@@ -339,4 +339,55 @@ mod tests {
             Ok(())
         );
     }
+
```

Comment:
I was slightly horrified to see how complicated the test case above is. I've tested the code more directly (i.e. without the actor actually running, just polling `next_request` manually) as a result. A better solution might be to try and abstract away some of the setup of the above test, and then run essentially the same test just with fewer in-flight requests to test the correct code path.